### PR TITLE
Filter panel header

### PR DESCRIPTION
**Pull request by maurya-k**

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -76,7 +76,8 @@
     "flair-visualizations": "master",
     "datatables.net-dt": "^3.2.2",
     "underscore": "^1.10.2",
-    "sql-formatter": "^1.1.2"
+    "sql-formatter": "^1.1.2",
```
Comment (khan-j) [inline bower.json:79]: there is no change why to update this file...just checkout this file

File: src/main/webapp/app/entities/flair-bi/filter/filter-penal-pin.component.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<form class="form-inline pin-filter" >
+    <div class="form-group" ng-repeat="dimension in vm.dimensions">
+        <div class="filter-section">
+            <label class="dimension-label">{{dimension.name}}</label>
+            <div class="dimensionFilter" ng-dropdown-multiselect="" options="vm.list[dimension.name]" selected-model="vm.selectedFilter[dimension.name]"
+                checkboxes="true" selectedToTop="true" extra-settings="vm.setting1"></div>
```
Comment (khan-j) [inline src/main/webapp/app/entities/flair-bi/filter/filter-penal-pin.component.html:6]: this is not valid name vm.setting1...can u rename to something suitable to functionality

File: src/main/webapp/app/entities/flair-bi/filter/filter-penal-pin.component.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
+(function () {
```
Comment (khan-j) [inline src/main/webapp/app/entities/flair-bi/filter/filter-penal-pin.component.js:1]: rename the file penal to panel

File: src/main/webapp/app/entities/flair-bi/filter/filter-penal-pin.component.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        vm.list = {};
+        vm.selectedFilter = {};
+        activate();
+
+        ////////////////
+
+        function activate() {
+
+            vm.setting1 = {
+                scrollableHeight: '200px',
```
Comment (khan-j) [inline src/main/webapp/app/entities/flair-bi/filter/filter-penal-pin.component.js:32]: rename to vm.setting1 to vm.settingStyle

File: src/main/webapp/app/entities/flair-bi/filter/filter-penal-pin.component.js [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+            $rootScope.$broadcast('flairbiApp:filter-add');
+            //addFilterInIframeURL();
+            $rootScope.$broadcast("flairbiApp:filterClicked");
+            $rootScope.$broadcast("flairbiApp:add-filter-In-FinterPenal");
+
+        }
+
+        function addFilterInIframeURL() {
+            var filters = filterParametersService.getSelectedFilter();
+            filterParametersService.setFilterInIframeURL(filters, vm.iframes, vm.dimensions);
```
Comment (khan-j) [inline src/main/webapp/app/entities/flair-bi/filter/filter-penal-pin.component.js:173]: no need to pass filters..its already a part of service

File: src/main/webapp/app/entities/flair-bi/filter/header-pin-filter.service.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+(function () {
+    'use strict';
+
+    angular
+        .module('flairbiApp')
+        .factory('headerPinFilterService', headerPinFilterService);
+
+    headerPinFilterService.$inject = ['$http'];
```
Comment (khan-j) [inline src/main/webapp/app/entities/flair-bi/filter/header-pin-filter.service.js:8]: this service is not required...can you place the http call in features service

File: src/main/webapp/app/entities/flair-bi/flair-bi-content-header.controller.js [hunk 7]
```diff
@@ -214,6 +219,24 @@
...
+            vm.pinDimensions = vm.dimensions.filter(function (item) {
+                return item.pin === true;
+            });
+            var headerSettings = {
+                showFSFilter: filterParametersService.getFiltersCount() == 0 ? false : true,
+                showPinFilter: vm.pinDimensions.length == 0 ? false : true
+            }
+            if (vm.pinDimensions.length > 0) {
+                vm.showPinFilter = true;
+                $rootScope.$broadcast("flairbiApp:toggle-headers-filters", headerSettings);
```
Comment (khan-j) [inline src/main/webapp/app/entities/flair-bi/flair-bi-content-header.controller.js:232]: $rootScope.$broadcast("flairbiApp:toggle-headers-filters", headerSettings); is duplicate. its being call in both the case

File: src/main/webapp/app/entities/flair-bi/flair-bi-content-header.controller.js [hunk 8]
```diff
@@ -421,12 +448,17 @@
         }
 
         function hideFiltersHeaderAndSideBar() {
```
Comment (khan-j) [inline src/main/webapp/app/entities/flair-bi/flair-bi-content-header.controller.js:450]: looks like this function is used 2 times....can be one generic function

File: src/main/webapp/app/entities/flair-bi/flair-bi-content-header.html [hunk 9]
```diff
@@ -12,7 +12,8 @@
             <form class="form-inline">
                 <div class="form-group">
                     <div class="dropdown">
-                        <button ng-disabled="vm.isIframe" class="btn btn-primary dropdown-toggle" type="button" data-toggle="dropdown">
+                        <button ng-disabled="vm.isIframe" class="btn btn-primary dropdown-toggle" type="button"
```
Comment (khan-j) [inline src/main/webapp/app/entities/flair-bi/flair-bi-content-header.html:15]: is there any code changed?

File: src/main/webapp/app/entities/flair-bi/flair-bi-content-header.html [hunk 10]
```diff
@@ -195,6 +197,13 @@
     </div>
 
 
+    <div class="row headerMenu" ng-if="vm.showPinFilter == true">
+        <div class="col-md-12 col-sm-6 col-xs-6 header-first">
+            <filter-penal-pin-component  view="vm.view" dimensions="vm.pinDimensions">
```
Comment (khan-j) [inline src/main/webapp/app/entities/flair-bi/flair-bi-content-header.html:202]: rename pinDimensions to pinedDimensions

File: src/main/webapp/app/entities/flair-bi/flair-bi.controller.js [hunk 11]
```diff
@@ -223,6 +225,25 @@
             $scope.$on("$destroy", unsubscribe);
         }
 
+        function showPinFilter() {
+            vm.pinDimensions = vm.dimensions.filter(function (item) {
+                return item.pin === true
+            });;
+            var headerSettings = {
```
Comment (khan-j) [inline src/main/webapp/app/entities/flair-bi/flair-bi.controller.js:232]: line 232 to 242 is duplicate code

File: src/main/webapp/app/entities/flair-bi/rightnavbar/flair-bi-rightnavbar.controller.js [hunk 12]
```diff
@@ -62,6 +62,7 @@
         vm.getSelectedItem = getSelectedItem;
         vm.onWidgetsOpen = onWidgetsOpen;
         vm.onWidgetsClose = onWidgetsClose;
+        vm.pinDimensions = [];
```
Comment (khan-j) [inline src/main/webapp/app/entities/flair-bi/rightnavbar/flair-bi-rightnavbar.controller.js:65]: it should pinedDimensions

File: src/main/webapp/app/entities/flair-bi/rightnavbar/flair-bi-rightnavbar.controller.js [hunk 13]
```diff
@@ -318,6 +332,33 @@
             $scope.$on("$destroy", unsubscribe);
         }
 
+        function registerAddFilterInFinterPenal() {
```
Comment (khan-j) [inline src/main/webapp/app/entities/flair-bi/rightnavbar/flair-bi-rightnavbar.controller.js:335]: penal to be renamed to panel;

File: src/main/webapp/app/entities/flair-bi/rightnavbar/flair-bi-rightnavbar.controller.js [hunk 14]
```diff
@@ -318,6 +332,33 @@
...
+            var unsubscribe = $scope.$on(
+                "flairbiApp:add-filter-In-FinterPenal",
+                function () {
+                    var filterParameters;
+                    filterParameters = filterParametersService.get();
+                    var filter = Object.keys(filterParameters);
+
+                    filter.forEach(element => {
+                        vm.dimensions.map(function (item) {
```
Comment (khan-j) [inline src/main/webapp/app/entities/flair-bi/rightnavbar/flair-bi-rightnavbar.controller.js:344]: do we need to loops...can you do this using single loop foreach (key.val) ot filter function

File: src/main/webapp/content/scss/partials/_filter-pane.scss [hunk 15]
```diff
@@ -30,6 +30,12 @@
   .favouritefilter {
     margin: auto;
   }
+  i.favouritefilter.fa.fa-star{
+      color: $blue;
```
Comment (khan-j) [inline src/main/webapp/content/scss/partials/_filter-pane.scss:34]: remove blue var color with primary_color..
pls note that $primary_color always use this color as it is primary color of app

File: src/main/webapp/content/scss/partials/_filter-pane.scss [hunk 16]
```diff
@@ -30,6 +30,12 @@
...
+  }
+  i.activePin{
+      color: $blue;
```
Comment (khan-j) [inline src/main/webapp/content/scss/partials/_filter-pane.scss:37]: here also

File: src/main/webapp/content/scss/partials/_flair-bi-content-header.scss [hunk 17]
```diff
@@ -375,6 +375,61 @@
         padding: 0 8px;
         margin-top: 0px;
     }
+    .headerMenu{
+        height: 35px;
+        display: flex;
+        border-bottom: 1px #e1e3e3 solid;
```
Comment (khan-j) [inline src/main/webapp/content/scss/partials/_flair-bi-content-header.scss:381]: use scss var for color e1e3e3

File: src/main/webapp/index.html [hunk 18]
```diff
@@ -483,13 +483,6 @@
             <polygon
                 points="298.667,30.187 268.48,0 149.333,119.147 30.187,0 0,30.187 119.147,149.333 0,268.48 30.187,298.667 149.333,179.52 268.48,298.667 298.667,268.48 179.52,149.333 " />
         </symbol>
-        <symbol id="upload-logo" viewBox="0 0 100 100" style="width:100%;height:100%;background-size:initial;background-repeat-y:initial;background-repeat-x:initial;background-position-y:initial;background-position-x:initial;background-origin:initial;background-image:initial;background-color:#fff;background-clip:initial;background-attachment:initial;animation-play-state:paused" xmlns="http://www.w3.org/2000/svg">
```
Comment (khan-j) [inline src/main/webapp/index.html:486]: looks like u have removed logo which is used for client logo in admin...pls add it back

**Review by khan-j — CHANGES_REQUESTED** [review]

File: bower.json (hunk 1 above)
Comment (maurya-k) [reply bower.json:79]: I have added angularjs-dropdown-multiselect package that's why its added comma

File: src/main/webapp/app/entities/flair-bi/filter/filter-element-grpc.component.js [hunk 19]
```diff
@@ -147,7 +149,7 @@
                     vm.dimension.favouriteFilter = !vm.dimension.favouriteFilter;
                     var opration = vm.dimension.favouriteFilter === true ? 'added to' : 'removed from';
                     var info = {
-                        text: "Dimensions " + opration + " Bookmark filter panelr",
+                        text: "Dimensions " + opration + " Bookmark filter panel",
```
[the remark this answers is not in the document]
Comment (maurya-k) [reply src/main/webapp/app/entities/flair-bi/filter/filter-element-grpc.component.js]: done

File: src/main/webapp/app/entities/flair-bi/filter/filter-element-grpc.component.js [hunk 20]
```diff
@@ -444,5 +446,25 @@
         function getSeparator(){
             return vm.separator ? vm.separator.value : ",";
         }
+        function toggleHeaderFilter(dimension) {
+            headerPinFilterService.markPinFilter(dimension.id, !vm.dimension.pin)
+                .then(function (data) {
+                    vm.dimension.pin = !vm.dimension.pin;
+                    var opration = vm.dimension.pin === true ? 'added to' : 'removed from';
+                    var info = {
+                        text: "Dimensions " + opration + " pin filter panel",
```
[the remark this answers is not in the document]
Comment (maurya-k) [reply src/main/webapp/app/entities/flair-bi/filter/filter-element-grpc.component.js]: done

File: src/main/webapp/app/entities/flair-bi/filter/filter-penal-pin.component.html [hunk 21]
```diff
@@ -0,0 +1,13 @@
+<form class="form-inline pin-filter" >
```
[the remark this answers is not in the document]
Comment (maurya-k) [reply src/main/webapp/app/entities/flair-bi/filter/filter-penal-pin.component.html]: done

File: src/main/webapp/app/entities/flair-bi/filter/filter-penal-pin.component.js (hunk 4 above)
Comment (maurya-k) [reply src/main/webapp/app/entities/flair-bi/filter/filter-penal-pin.component.js]: done

File: src/main/webapp/app/entities/flair-bi/filter/filter-penal-pin.component.js (hunk 5 above)
Comment (maurya-k) [reply src/main/webapp/app/entities/flair-bi/filter/filter-penal-pin.component.js]: done

File: src/main/webapp/app/entities/flair-bi/filter/header-pin-filter.service.js (hunk 6 above)
Comment (maurya-k) [reply src/main/webapp/app/entities/flair-bi/filter/header-pin-filter.service.js]: done

File: src/main/webapp/app/entities/flair-bi/flair-bi-content-header.html (hunk 9 above)
Comment (maurya-k) [reply src/main/webapp/app/entities/flair-bi/flair-bi-content-header.html:15]: done

File: src/main/webapp/app/entities/flair-bi/flair-bi-content-header.html (hunk 10 above)
Comment (maurya-k) [reply src/main/webapp/app/entities/flair-bi/flair-bi-content-header.html]: done

File: src/main/webapp/app/entities/flair-bi/flair-bi.controller.js (hunk 11 above)
Comment (maurya-k) [reply src/main/webapp/app/entities/flair-bi/flair-bi.controller.js:232]: done

File: src/main/webapp/app/entities/flair-bi/rightnavbar/flair-bi-rightnavbar.controller.js (hunk 12 above)
Comment (maurya-k) [reply src/main/webapp/app/entities/flair-bi/rightnavbar/flair-bi-rightnavbar.controller.js]: done

File: src/main/webapp/app/entities/flair-bi/rightnavbar/flair-bi-rightnavbar.controller.js (hunk 13 above)
Comment (maurya-k) [reply src/main/webapp/app/entities/flair-bi/rightnavbar/flair-bi-rightnavbar.controller.js]: done

File: src/main/webapp/content/scss/partials/_filter-pane.scss (hunk 15 above)
Comment (maurya-k) [reply src/main/webapp/content/scss/partials/_filter-pane.scss]: done

File: src/main/webapp/content/scss/partials/_filter-pane.scss (hunk 16 above)
Comment (maurya-k) [reply src/main/webapp/content/scss/partials/_filter-pane.scss]: done

**Review by khan-j — APPROVED** [review]

**Review by khan-j — DISMISSED** [review]

**Pull request merged**